File: src/main/java/com/kyiv_kug/modificator/SuperClass.java
package com.kyiv_kug.modificator;

public class SuperClass {
    protected void protectedMethod(){
        System.out.println("SuperClass#protectedMethod");
    }
}

class ChildClass extends SuperClass{

    private SuperClass superClass;

    public ChildClass(SuperClass superClass) {
        this.superClass = superClass;
    }

    @Override
    protected void protectedMethod() {
        super.protectedMethod();
        superClass.protectedMethod();
    }
}

File: src/main/java/com/kyiv_kug/modificator/solution/SuperClass.kt
package com.kyiv_kug.modificator.solution

open class SuperClass {
    /**
     * cannot be protected in the example
     * https://youtrack.jetbrains.com/issue/KT-21048
     */
    internal open fun protectedMethod() {
        println("SuperClass#protectedMethod")
    }
}

class ChildClass(
        private val superClass: SuperClass
) : SuperClass() {

    override fun protectedMethod() {
        super.protectedMethod()
        superClass.protectedMethod()
    }
}

fun main(args: Array<String>) {
    ChildClass(
            SuperClass()
    ).protectedMethod()
}